app/actions/local/channel.test.ts

```diff
@@ -0,0 +1,468 @@
+
+        const listenerCallback = jest.fn();
+        const listener = DeviceEventEmitter.addListener(Navigation.NAVIGATION_HOME, listenerCallback);
+        const {models, error} = await switchToChannel(serverUrl, channelId);
+        listener.remove();
+
+        const systemValues = await queryCommonSystemValues(operator.database);
+        const teamHistory = await queryTeamHistory(operator.database);
+        const channelHistory = await queryChannelHistory(operator.database, teamId);
+        const member = await queryMyChannel(operator.database, channelId);
+
+        expect(error).toBeUndefined();
+        expect(models?.length).toBe(3); // Viewed at, channel history and currentChannelId
+        expect(systemValues.currentTeamId).toBe(teamId);
+        expect(systemValues.currentChannelId).toBe(channelId);
+        expect(teamHistory.length).toBe(0);
+        expect(channelHistory.length).toBe(1);
+        expect(channelHistory[0]).toBe(channelId);
+        expect(member?.lastViewedAt).toBe(now);
+        expect(dismissAllModalsAndPopToScreen).toHaveBeenCalledTimes(1);
+        expect(dismissAllModalsAndPopToRoot).toHaveBeenCalledTimes(0);
+        expect(listenerCallback).toHaveBeenCalledTimes(0);
+    });
+
+    it('switch to a DM without passing the teamId', async () => {
+        await operator.handleTeam({teams: [team], prepareRecordsOnly: false});
+        const tChannel = {...channel, team_id: ''};
+        await operator.handleChannel({channels: [tChannel], prepareRecordsOnly: false});
+        await operator.handleMyChannel({channels: [tChannel], myChannels: [channelMember], prepareRecordsOnly: false});
+        await operator.handleSystem({systems: [{id: SYSTEM_IDENTIFIERS.CURRENT_TEAM_ID, value: teamId}, {id: SYSTEM_IDENTIFIERS.CURRENT_CHANNEL_ID, value: 'currentChannelId'}], prepareRecordsOnly: false});
+
+        const listenerCallback = jest.fn();
+        const listener = DeviceEventEmitter.addListener(Navigation.NAVIGATION_HOME, listenerCallback);
+        const {models, error} = await switchToChannel(serverUrl, channelId);
+        listener.remove();
+
+        const systemValues = await queryCommonSystemValues(operator.database);
+        const teamHistory = await queryTeamHistory(operator.database);
+        const channelHistory = await queryChannelHistory(operator.database, teamId);
+        const member = await queryMyChannel(operator.database, channelId);
+
+        expect(error).toBeUndefined();
+        expect(models?.length).toBe(3); // Viewed at, channel history and currentChannelId
+        expect(systemValues.currentTeamId).toBe(teamId);
+        expect(systemValues.currentChannelId).toBe(channelId);
+        expect(teamHistory.length).toBe(0);
+        expect(channelHistory.length).toBe(1); // Currently errors since it is being stored in the current team history and not the destination history
```

Comment:
@larkox Not sure why the comment says it currently errors out but when I run the test it passes.

Reply:
Used to be like that but now is solved. I thought I had removed that comment already :sweat_smile: